packages/docs/content/docs/prompting-guideline.mdx

```diff
@@ -0,0 +1,149 @@
+- **Leverage Pochi Features:**
+  - Use `~/.pochi/config.jsonc` for [models](/models) & [MCPs](/mcp).
+  - Use `README.pochi.md` for [project-specific rules](/rules).
+  - Use `.pochi/workflows/` for [custom workflows](/workflows).
+  - Use `.pochi/agents/` for [custom agents](/custom-agent).
+  - Use [Pochi's sharing features](/share) to share task histories.
+  - Use the [GitHub](/github) & [Slack](/slack) integrations for team collaboration.
+
+## How to Prompt Pochi
+
+Effective prompting transforms vague ideas into precise actions by emphasizing clarity, examples, and structure. Approach it as directive a collaborative agent that benefits from guided, step-by-step instructions to deliver reliable results.
+
+### Craft Clear Prompts
+
+Be specific and direct. Avoid indirect phrasing like “can you", state the action explicitly.
+
+**Example:** “Fix `@utils.js` line 42: Add input validation for `process(data)` where `data` can be null, return a diff.”
+
+### Use Few Shot Prompts
+
+Provide 2-3 input-output examples to set expectations
+
+**Example:**
+“Input: Error in `@hooks.js` - ‘useEffect is not defined’ in `render()`;
+Output: `import { useEffect } from 'react'; useEffect(() => { console.log('mounted'); }, []);`.”
+
+**Example:**
+ “Input: Slow `@api.js` - 5s load on `/users` due to DB query;
+Output: `const cache = require('node-cache'); cache.set('users', await db.query('SELECT * FROM users'), 3600);`.”
+
+### Let Pochi Think (Chain-of-Thought)
+
+For complex logic, request step-by-step reasoning.
+
+**Example:**
+“Debug `@main.py`: Identify the null reference issue in `fetchUser()`, propose a fix, test with `pytest test_main.py`, log the result."
+
+### Use Structured Prompts
+
+Apply structured formats (XML or JSON) for precision, repeatability, and task chaining. Use XML for hierarchical edits (e.g., nested tags) or JSON for data-driven tasks (e.g., key-value pairs).
+
+**XML:**
+```xml
+<edit>
+  <file>@api.js</file>
+  <line>42</line>
+  <context>function validateUser(input) { return input.length; } in @api.js fails with TypeError: Cannot read property 'length' of null</context>
+  <fix>
+    try {
+      if (!input) throw new TypeError('Input cannot be null');
+      return validateInput(input);
+    } catch (e) {
+      logError(e, { file: '@api.js', line: 42 });
+      return false;
+    }
+  </fix>
+</edit>
+```
+
+**JSON:**
+```json
+{
+  "task": "refactor",
+  "file": "@api.js",
+  "line": 42,
+  "context": "validateUser(input) fails with TypeError: Cannot read property 'length' of null",
+  "change": "add input validation with try-catch"
+}
+```
+
+### Assign Pochi a Role (System Prompts)
+
+Specify a role in the prompt or via a [custom agent](/custom-agent) to enforce standards.
+
+**Example:**
+“Act as linter for `@models/db.py` per `README.pochi.md`, focusing on PEP 8 spacing issues.”
```

Comment:
this example doesn't show case custom agent. and README.pochi.md doesn't really need explicit reference